src/dependencies.py

```diff
@@ -1,28 +1,39 @@
-from typing import Annotated
+from functools import lru_cache
+from src.configs import Settings
+from src.repositories import MoleculeRepository
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from src.service import MoleculeService
+from src.schemas import PaginationQueryParams
 
-from fastapi import HTTPException, Query
-from fastapi import status
-from .repository.molecule_repositories import AbstractMoleculeRepository, HTTPMoleculeRepository
-from .config import Config
 
+@lru_cache
+def get_database_url():
+    return Settings().database_url
 
-def get_config() -> Config:
-    if not hasattr(get_config, "config"):
-        get_config.config = Config()
-    return get_config.config
 
+@lru_cache
+def get_session_factory():
+    if Settings().TEST_MODE:
+        return sessionmaker(
+            bind=create_engine(
+                get_database_url(), connect_args={"check_same_thread": False}
+            )
+        )
```

Comment:
Instead of having session as a dependency, I decided to have sessionmaker, and I pass this to 
repository class.